pkg/sql/compile/compile.go

```diff
@@ -421,6 +421,9 @@ func (c *Compile) compileQuery(ctx context.Context, qry *plan.Query) ([]*Scope,
 	if err != nil {
 		return nil, err
 	}
+	if plan2.IsTpQuery(qry, int32(len(c.cnList))) {
```

Comment:
why not set the plan flag in plan but in compile..

Reply:
fixed